### ISSUE TITLE
Timezone scopeDate method does not respect interface

### ISSUE THREAD

**Issue by sta1r** [magento/magento2#23359]
### Preconditions (*)

1. 2.3-develop

### Steps to reproduce (*)

1. Start with a class similar to this:
```
<?php

class MyDate
{
    /**
     * @var \Magento\Framework\Stdlib\DateTime\TimezoneInterface
     */
    private $timeZone;

    /**
     * @param \Magento\Framework\Stdlib\DateTime\TimezoneInterface $timezone
     */
    public function __construct(
        \Magento\Framework\Stdlib\DateTime\TimezoneInterface $timezone
    ) {
        $this->timeZone = $timezone;

        $scopedDate = $this->timeZone->scopeDate(1, new \DateTime, true);
    }
}
```
2. Run the code

### Expected result (*)

1. `scopeDate()` returns a DateTime scoped to the store 1 locale. 
2. The [method](https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php#L196-L205) should work according to signature in the  [interface](https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/Stdlib/DateTime/TimezoneInterface.php#L73-L81) - support all types

### Actual result (*)

1. PHP Fatal error is thrown
```
PHP Fatal error:  Uncaught TypeError: DateTime::__construct() expects parameter 1 to be string, object given
```

The `scopeDate()` [method](https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php#L196-L205) is not implementing its [interface](https://github.com/magento/magento2/blob/2.3-develop/lib/internal/Magento/Framework/Stdlib/DateTime/TimezoneInterface.php#L73-L81) (or at least the docBlock of the interface, which indicates this method can take a DateTime as `$date`).

**Comment by sta1r** [comment]
Should it take a DateTime or not? Happy to submit a fix either way.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sdzhepa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18044` were created

**Issue Available**: @sdzhepa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by edenduong** [comment]
I think we should implement it like date(..) function in the same class:

https://github.com/magento/magento2/pull/23693

**Comment by magento-engcom-team** [comment]
Hi @sta1r. Thank you for your report. 
The issue has been fixed in magento/magento2#23693 by @edenduong in 2.3-develop branch
Related commit(s):
 - [4b7f8e5544479087a32cfbfe1c4f5a116a208e53](https://github.com/magento/magento2/commit/4b7f8e5544479087a32cfbfe1c4f5a116a208e53)

The fix will be available with the upcoming 2.3.4 release.
